test/integration/api/order.spec.js

```diff
@@ -656,6 +656,246 @@ test('updates an order move', async (t) => {
+
+  // receive payment confirmation (server-side)
+  const updatedOrder = await payOrder(order)
+
+  t.true(updatedOrder.amountRemaining === 0)
+
+  // ///////////////// //
+  // HELPERS FUNCTIONS //
+  // ///////////////// //
+
+  function getCart (user) {
+    return _.get(user, 'metadata._private.cart', [])
+  }
+
+  async function saveCartIntoUser (user, cart) {
+    const { body: updatedUser } = await request(t.context.serverUrl)
+      .patch(`/users/${user.id}`)
+      .send({
+        metadata: {
+          _private: { cart }
+        }
+      })
+      .set(userHeaders)
+      .expect(200)
+
+    return updatedUser
+  }
+
+  async function updateCart (user, { assetId, quantity = 1 }) {
+    let cart = getCart(user)
+
+    if (quantity === 0) {
+      cart = cart.filter(l => l.assetId !== assetId)
+    } else {
+      const line = cart.find(l => l.assetId === assetId)
+
+      if (line) {
+        cart = cart.map(l => {
+          if (l.assetId === assetId) return { assetId, quantity }
+          else return l
+        })
+      } else {
+        cart = cart.concat([{ assetId, quantity }])
+      }
+    }
+
+    return saveCartIntoUser(user, cart)
+  }
+
+  async function getPreviewedTransactions (user) {
+    const cart = getCart(user)
+    const previewedTransactions = []
+
+    for (const cartLine of cart) {
+      const { body: preview } = await request(t.context.serverUrl)
+        .post('/transactions/preview')
+        .send(cartLine)
+        .set(userHeaders)
+        .expect(200)
+
+      previewedTransactions.push(preview)
+    }
+
+    return previewedTransactions
+  }
+
+  async function createTransactionsFromCart (user) {
+    const cart = getCart(user)
+    const transactions = []
+
+    for (const cartLine of cart) {
+      const { body: transaction } = await request(t.context.serverUrl)
+        .post('/transactions')
+        .send(cartLine)
+        .set(userHeaders)
+        .expect(200)
+
+      transactions.push(transaction)
+    }
+
+    return transactions
+  }
+
+  async function createOrderFromTransactions (transactions) {
+    const { body: order } = await request(t.context.serverUrl)
+      .post('/orders')
+      .send({
+        transactionIds: transactions.map(t => t.id)
+      })
+      .set(userHeaders)
+      .expect(200)
+
+    return order
+  }
+
+  async function payOrder (order) {
+    const transactionIds = _.uniq(order.lines.map(l => l.transactionId))
+
+    for (const transactionId of transactionIds) {
+      await request(t.context.serverUrl)
+        .post(`/transactions/${transactionId}/transitions`)
+        .send({ name: 'pay' })
+        .set(authorizationHeaders)
+        .expect(200)
+    }
+
+    await request(t.context.serverUrl)
+      .post('/order-moves')
+      .send({
+        orderId: order.id,
+        payerId: order.payerId,
+        payerAmount: order.amountDue,
+        currency: order.currency,
+      })
+      .set(authorizationHeaders)
+      .expect(200)
```

Comment:
Isn’t this supposed to be transactional in the technical sense?

What if the `/order-moves` request fails after transitions have been applied to transactions and we have an inconsistent state as a result?
I asked about a more integrated sequence on slack but maybe we didn’t spare enough time to think about this.

I don’t promote API complexity, but here we could help Order API user to feel safer and have less work to do to add transitions to every single transaction. Extra parameters might be worth it.

What about some utility parameter (like `transactionTransitionName`, set to a valid transition name) in Order moves API that could automatically add transition to all transactions given transactions `transactionIdsInTransition`, in a **Postgres transaction**.

It’s a feature so it’s not necessary to include such a change in this PR but this could make an issue and a PR on its own.

Reply:
Yes you're right, it is a transactional operation. Another PR is needed to add this feature.